nilearn/maskers/base_masker.py

```diff
@@ -66,6 +66,19 @@ def _filter_and_extract(
         print("[%s] Loading data from %s" % (
             class_name,
             _utils._repr_niimgs(imgs, shorten=False)))
+
+    imgs = _utils.check_niimg(imgs)
+
+    # Raise warning if a 3D niimg is provided.
+    if imgs.ndim == 3:
+        warnings.warn(
+            'Starting in version 0.12, 3D images will be transformed to '
+            '1D arrays. '
+            'Until then, 3D images will be coerced to 2D arrays, with a '
+            'singleton first dimension representing time.',
+            DeprecationWarning,
+        )
+
     imgs = _utils.check_niimg(imgs, atleast_4d=True, ensure_ndim=4,
                               dtype=dtype)
```

Comment:
`imgs` may be a string (or list of strings, I guess) at this point. I therefore needed to look at the Nifti1Image object first, _before_ changing its dimensionality with `check_niimg(..., ensure_ndim=4)`. The best solution I could come up with was to call `check_niimg` twice.